File: src/main/java/com/marknkamau/Exceptions/IncorrectInputExtension.java
package com.marknkamau.Exceptions;

public class IncorrectInputExtension extends BaseException {
    private static final String TITLE = "Incompatible input file";
    private static final String INPUT_EXTENSION_MESSAGE = "Input file must be a plain text file (.txt)";

    public IncorrectInputExtension() {
        super(TITLE, INPUT_EXTENSION_MESSAGE);
    }
}

File: src/main/java/com/marknkamau/Exceptions/IncorrectInputExtension.kt
package com.marknkamau.Exceptions

class IncorrectInputExtension : BaseException(IncorrectInputExtension.TITLE, IncorrectInputExtension.INPUT_EXTENSION_MESSAGE) {
    companion object {
        private val TITLE = "Incompatible input file"
        private val INPUT_EXTENSION_MESSAGE = "Input file must be a plain text file (.txt)"
    }
}
